### PR TITLE
Aggregate parquet files

### PR DESCRIPTION
**Pull request by fabyan-alexander**

### CODE REVIEW

**Comment by fabyan-alexander** [comment]
[[BillingAg] Aggregate parquet files in a new parquet file.](https://trello.com/c/WiQknqdv/969-billingag-aggregate-parquet-files-in-a-new-parquet-file)

File: src/main/java/com/leaf/billing/FieldBoundaryAggregatorApp.java [hunk 1]
```diff
@@ -11,35 +22,49 @@ public class FieldBoundaryAggregatorApp {
      * @param args
      */
     public static void main(String[] args) throws IOException {
+        JobProperties jobProperties = PropertyBuilder.build(args);
+        try (SparkSession sparkSession = SparkSession.builder()
+                .appName("BillingAggregator2")
```
Comment (rgdafonso) [inline src/main/java/com/leaf/billing/FieldBoundaryAggregatorApp.java:27]: What happened to BillingAgregator1?

**Review by rgdafonso — APPROVED** [review]
Nice

File: src/main/java/com/leaf/billing/utils/SparkCommands.java [hunk 2]
```diff
@@ -31,4 +36,43 @@ public static LocalDate locateOldestTimestampGroup(String path) throws IOExcepti
         return currentOldest;
     }
 
+    public static Dataset<Row> loadParquetFilesBasedOnDate(SparkSession sparkSession, String folderPath,
```
Comment (ricardosantacattarina) [inline src/main/java/com/leaf/billing/utils/SparkCommands.java:39]:
```suggestion
    public static Dataset<Row> loadParquetFilesForSpecifiedDate(SparkSession sparkSession, String folderPath,
```
I took me a little bit to understand that this is for the specific date, not for a range or before/after the timestamp.

File: src/main/java/com/leaf/billing/utils/SparkCommands.java [hunk 3]
```diff
@@ -31,4 +36,43 @@ public static LocalDate locateOldestTimestampGroup(String path) throws IOExcepti
...
+     * Writes a dataset into a normalized timestamp folder.
+     * The normalized folder is BASEPATH/YYYY/MM/DD/YYYY-MM-DD.parquet
+     *
+     * @param targetBaseFolder base folder, can be a local files system or a S3 filesystem
+     * @param timestamp        used to create the classified folders
+     * @param dataset          data that will be written in the new file
+     *
+     * @return the location of the file created in the targetBaseFolder
+     */
+    public static String writeFileOnTimestampFolder(String targetBaseFolder, LocalDate timestamp,
```
Comment (ricardosantacattarina) [inline src/main/java/com/leaf/billing/utils/SparkCommands.java:56]:
```suggestion
    public static String writeFileOnDateFolder(String targetBaseFolder, LocalDate timestamp,
```

**Review by ricardosantacattarina — APPROVED** [review]
Nice!

**Pull request merged**